charts/MapTooltip.tsx

```diff
@@ -89,7 +89,8 @@ export class MapTooltip extends React.Component<MapTooltipProps> {
 
     @computed get barColor() {
         const { colorScale } = this.chart.map.data
-        return colorScale.singleColorScale && !colorScale.isCustomColors
+        return colorScale.singleColorScale &&
+            !colorScale.customNumericColorsActive
```

Comment:
It'd be nice to keep a "isCustomColors" function, so that the implementation detail of numeric vs categorical doesn't have to be considered for the developer just wondering if the color scales are automatic or manually overridden.

Reply:
Yeah I was looking at this – I only renamed `isCustomColors`, the implementation is the same. For maps, in 99% of cases we use a numerical legend ([this](https://ourworldindata.org/grapher/continents-according-to-our-world-in-data) is what a categorical legend looks like).

But, there is one categorical bin always, even on a numerical legend – the "No data" bin. I thought we'd like to exclude that from the logic here anyway, it makes sense to only consider numerical bins when determining the bar color.

Let me know if you disagree!

Reply:
If we want to have a property that considers both we can add this:

```ts
    @computed get hasCustomColors(): boolean {
        return (
            (this.customNumericColorsActive &&
                this.customNumericColors.some(color => color !== undefined)) ||
            Object.values(this.customCategoryColors).some(
                color => color !== undefined
            )
        )
    }
```

But I don't think we should add it unless we use it in some case.

Reply:
I think this is actually the use case for that function. `barColor` is basically asking, "Is this a non-divergent/multi-hue color scale?", which cannot be answered with certainty for any color scheme with a manual override. (I manually vetted all the ColorBrewer, etc. color schemes 😄   to specify `singleColorScheme`)

Reply:
But we can't create a bar plot with categorical values (categorical means they are type `string`), so it doesn't matter if those have custom colors? And as I mentioned, _No data_ is also a categorical bin that always exists, but we want to ignore it even if it has a custom color? So any categorical bins don't matter in this case?

Reply:
Ah, ok. Ideally I'd be able to do something like 
```
renderBars(): 
    return colorScale.isNumeric

barColor():
    return colorScale.hasCustomColors ? NEUTRAL_COLOR : colorScale.darkestColor
```
All of which would ignore the "No Data" bin.

But this is just a preference so happy to leave this as-is for now!

Reply:
:+1: The implementation here is unchanged apart from the renamed prop, we can make the change outside this PR.